src/scripts/create_norglm_multiqa.py

```diff
@@ -0,0 +1,203 @@
+
+        if answer.lower() in context.lower():
+            return answer
+
+        # Use OpenAI to rephrase the answer
+        chat_completion = client.chat.completions.create(
+            messages=[
+                {
+                    "role": "user",
+                    "content": (
+                        f"Given the following context: '{context}', and the '{question}' please rephrase the answer, so that it matches exactly "
+                        f"with a phrase from the context in a case-insensitive way. E.g. it must be possible to find the rephrased answer "
+                        f"in the context without any modifications. "
+                        f"The rephrased answer should be as concise as possible, preferable not more than 7 words, and ideally 3 or less words."
+                        f"Now rephrase this answer: '{answer}'"
+                    ),
+                }
+            ],
+            model="gpt-4o",
+            seed=4242,
+            temperature=0,
+        )
+
+        rephrased_answer = chat_completion.choices[0].message.content
+        return rephrased_answer
+
+    # Use the original answer if it could be found in the context
+    df_orig_bool = df[["answer", "context"]].apply(
+        lambda row: row["answer"].lower() in row["context"].lower(), axis=1
+    )
+    df_orig = df[df_orig_bool]
+
+    # For the rest of the answers, we try to rephrase them
+    df_no_context = df[~df_orig_bool]
+
+    # Rephrase the answers
+    df_no_context.loc[:, "answer"] = df_no_context.apply(
+        lambda row: rephrase_answer(row["question"], row["answer"], row["context"]),
+        axis=1,
+    )
+
+    # Remove non-word start and end characters from answers
+    df_no_context.loc[:, "answer"] = df_no_context["answer"].str.replace(
+        "(^\W|\W$)", "", regex=True
+    )
+
+    # Only keep the rephrased answers where the answer could be found in the context
+    df_in_context = df_no_context[["answer", "context"]].apply(
+        lambda row: row["answer"].lower() in row["context"].lower(), axis=1
+    )
+    df_with_answer = df_no_context[df_in_context]
+
+    # Combine original with the rephrased answers
+    cleaned_df = pd.concat([df_orig, df_with_answer])
+
+    cleaned_df = cleaned_df.reset_index(drop=True)
+
+    # Convert to 'answers' style column
+    cleaned_df.loc[:, "answers"] = cleaned_df.apply(
+        lambda row: {
+            "text": [row["answer"]],
+            "answer_start": row["context"].lower().index(row["answer"].lower()),
+        },
+        axis=1,
+    )
+
+    # Create validation split
+    val_size = 256
+    val_df = df.sample(n=val_size, random_state=4242)
+
+    # Create test split
+    test_size = 1024
+    filtered_df = df[~df.index.isin(val_df.index)]
+    test_df = filtered_df.sample(n=test_size, random_state=4242)
+
+    # Create train split
+    train_df = filtered_df[~filtered_df.index.isin(test_df.index)]
```

Comment:
How many samples are there? Maybe you can add a `assert len(cleaned_df) == XXX` before the splits, just to make it clear to the reader what's going on?

Reply:
Yeah makes sense! I also just noticed that I used the wrong df for splitting.. gonna run the rephrasing again and get the total 😄

Reply:
Should be fixed now!